winsup/cygwin/mm/malloc_wrapper.cc

```diff
@@ -50,13 +50,26 @@ import_address (void *imp)
 {
   __try
     {
+#if defined(__aarch64__)
+      uint32_t opcode = *(uint32_t *) imp;
+      // if opcode is an adr instruction
+      if ((opcode & 0x9f000000) == 0x10000000)
+	{
+	  uint64_t immhi = (opcode >> 5) & 0x3ffff;
```

Comment:
Where I can find some references, e.g. documentation or some implementation elsewhere, what does this calculation do?

Reply:
https://developer.arm.com/documentation/ddi0602/2024-12/Base-Instructions/ADR--Form-PC-relative-address-